packages/twenty-server/src/database/commands/0-22-update-boolean-fields-null-default-values-and-null-values.command.ts

```diff
@@ -0,0 +1,151 @@
+import { Repository } from 'typeorm';
+import chalk from 'chalk';
+
+import { DataSourceService } from 'src/engine/metadata-modules/data-source/data-source.service';
+import { Workspace } from 'src/engine/core-modules/workspace/workspace.entity';
+import { TypeORMService } from 'src/database/typeorm/typeorm.service';
+import {
+  FieldMetadataEntity,
+  FieldMetadataType,
+} from 'src/engine/metadata-modules/field-metadata/field-metadata.entity';
+import { WorkspaceCacheVersionService } from 'src/engine/metadata-modules/workspace-cache-version/workspace-cache-version.service';
+import { ObjectMetadataEntity } from 'src/engine/metadata-modules/object-metadata/object-metadata.entity';
+import { computeObjectTargetTable } from 'src/engine/utils/compute-object-target-table.util';
+
+interface UpdateBooleanFieldsNullDefaultValuesAndNullValuesCommandOptions {
+  workspaceId?: string;
+}
+
+@Command({
+  name: 'migrate-0.22:update-boolean-field-null-default-values-and-null-values',
+  description:
+    'Update boolean fields null default values and null values to false',
+})
+export class UpdateBooleanFieldsNullDefaultValuesAndNullValuesCommand extends CommandRunner {
+  private readonly logger = new Logger(
+    UpdateBooleanFieldsNullDefaultValuesAndNullValuesCommand.name,
+  );
+  constructor(
+    @InjectRepository(Workspace, 'core')
+    private readonly workspaceRepository: Repository<Workspace>,
+    @InjectRepository(FieldMetadataEntity, 'metadata')
+    private readonly fieldMetadataRepository: Repository<FieldMetadataEntity>,
+    @InjectRepository(ObjectMetadataEntity, 'metadata')
+    private readonly objectMetadataRepository: Repository<ObjectMetadataEntity>,
+    private readonly typeORMService: TypeORMService,
+    private readonly dataSourceService: DataSourceService,
+    private readonly workspaceCacheVersionService: WorkspaceCacheVersionService,
+  ) {
+    super();
+  }
+
+  @Option({
+    flags: '-w, --workspace-id [workspace_id]',
+    description: 'workspace id. Command runs on all workspaces if not provided',
+    required: false,
+  })
+  parseWorkspaceId(value: string): string {
+    return value;
+  }
+
+  async run(
+    _passedParam: string[],
+    options: UpdateBooleanFieldsNullDefaultValuesAndNullValuesCommandOptions,
+  ): Promise<void> {
+    let workspaceIds: string[] = [];
+
+    if (options.workspaceId) {
+      workspaceIds = [options.workspaceId];
+    } else {
+      workspaceIds = (await this.workspaceRepository.find()).map(
+        (workspace) => workspace.id,
+      );
+    }
+
+    if (!workspaceIds.length) {
+      this.logger.log(chalk.yellow('No workspace found'));
+
+      return;
+    } else {
+      this.logger.log(
+        chalk.green(`Running command on ${workspaceIds.length} workspaces`),
+      );
+    }
+
+    for (const workspaceId of workspaceIds) {
+      const dataSourceMetadatas =
+        await this.dataSourceService.getDataSourcesMetadataFromWorkspaceId(
```

Comment:
So technically datasource is a one-to-many but in practice you will always have 1 datasource per workspace, we are actually planning to remove the datasource table and move its content to the workspace table directly.
In your case you can simply use "getLastDataSourceMetadataFromWorkspaceId" which is what we usually use in the codebase so you can remove the for loop here to make this part easier.
In practice it does not change much so it's up to you!